src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/RowDefinitionCollectionConverter.cs

```diff
@@ -0,0 +1,117 @@
+    internal class RowDefinitionCollectionConverter : TypeConverter
+    {
+        #region Public Methods
+
+        /// <summary>
+        /// CanConvertFrom - Returns whether or not this class can convert from a given type.
+        /// </summary>
+        /// <returns>
+        /// bool - True if thie converter can convert from the provided type, false if not.
+        /// </returns>
+        /// <param name="context"> The ITypeDescriptorContext for this call. </param>
+        /// <param name="sourceType"> The Type being queried for support. </param>
+        public override bool CanConvertFrom(ITypeDescriptorContext context, Type sourceType)
+        {
+            return sourceType == typeof(string) || base.CanConvertFrom(context, sourceType);
+        }
+
+        /// <summary>
+        /// CanConvertTo - Returns whether or not this class can convert to a given type.
+        /// </summary>
+        /// <returns>
+        /// bool - True if this converter can convert to the provided type, false if not.
+        /// </returns>
+        /// <param name="context"> The ITypeDescriptorContext for this call. </param>
+        /// <param name="destinationType"> The Type being queried for support. </param>
+        public override bool CanConvertTo(ITypeDescriptorContext context, Type destinationType)
+        {
+            return destinationType == typeof(string) || base.CanConvertTo(context, destinationType);
+        }
+
+        /// <summary>
+        /// ConvertFrom - Attempt to convert to a RowDefinitionCollection from the given object.
+        /// </summary>
+        /// <returns>
+        /// The object which was constructoed.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        /// An ArgumentNullException is thrown if the example object is null.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        /// An ArgumentException is thrown if the object is not null and is not a valid type,
+        /// or if the destinationType isn't one of the valid destination types.
+        /// </exception>
+        /// <param name="context"> The ITypeDescriptorContext for this call. </param>
+        /// <param name="culture"> The CultureInfo which is respected when converting. </param>
+        /// <param name="value"> The Thickness to convert. </param>
+        public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo culture, object value)
+        {
+            if(value != null && value is string input){
+                IProvideValueTarget ipvt = context?.GetService(typeof(IProvideValueTarget)) as IProvideValueTarget;
+                Grid grid = ipvt?.TargetObject as Grid;
+                if(grid != null)
+                {
+                    var collection = new RowDefinitionCollection(grid); // Pass Grid instance
+                    var converter = new GridLengthConverter();
+
+                    TokenizerHelper th = new TokenizerHelper(input, culture);
```

Comment:
Not sure what's the timeframe on this getting merged, but #9364 but introduce a non-allocating tokenizer, which could be taken advantage of here already if this is merged after.

Reply:
@himgoyalmicro I believe you can resolve this, it will require to spanify the GridLengthConverter function as well, so that's imho out of scope, I'll just do it later when that PR is merged.

Reply:
Thank you @h3xds1nz !!